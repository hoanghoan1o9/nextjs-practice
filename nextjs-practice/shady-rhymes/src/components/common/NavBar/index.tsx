// Libraries
import React from 'react';
import { Stack, Popover, PopoverTrigger, Flex } from '@chakra-ui/react';

// Constants
import { APP_IMAGE_URLS } from '@constants/images';
import { ROUTES } from '@constants/router';

// Component
import Logo from '../Logo';
import { ChakraNextLink } from '@components/common/ChakraNextLink';

export interface NavItemProps {
  label: string;
  href?: string;
}

interface NavBarProps {
  navItems: NavItemProps[];
}

const NavBarBase = ({ navItems = [] }: NavBarProps) => {
  return (
    <Flex
      as="nav"
      data-testid="testNavbar"
      justifyContent="space-between"
      alignItems="center"
    >
      <Logo
        url={APP_IMAGE_URLS.LOGO}
        alt="main-logo"
        blurDataURL={APP_IMAGE_URLS.LOGO}
      />
      <Stack
        direction={'row'}
        spacing={4}
        display={{ base: 'none', lg: 'flex' }}
      >
        {navItems.map((navItem) => (
          <Flex key={navItem.label}>
            <Popover trigger={'hover'}>
              <PopoverTrigger>
                <ChakraNextLink
                  href={navItem.href ?? ROUTES.HOME_PAGE}
                  size="navBar"
                  variant="navBar"
                  ml="10"
                >
                  {navItem.label}
                </ChakraNextLink>
              </PopoverTrigger>
            </Popover>
          </Flex>
        ))}
      </Stack>
      <Logo
        url={APP_IMAGE_URLS.BAG_ICON}
        height={24}
        width={24}
        alt="bag-icon"
        blurDataURL={APP_IMAGE_URLS.BAG_ICON}
      />
    </Flex>
  );
};

export default NavBarBase;
