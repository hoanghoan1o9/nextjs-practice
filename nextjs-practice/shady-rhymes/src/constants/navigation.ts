// Types
export interface NavItemProps {
  label: string;
  href?: string;
}

export const NAV_ITEMS: NavItemProps[] = [
  {
    label: 'Furniture  ',
    href: '#',
  },
  {
    label: 'Shop',
    href: '#',
  },
  {
    label: 'About Us',
    href: '#',
  },
  {
    label: 'Contact',
    href: '#',
  },
];
