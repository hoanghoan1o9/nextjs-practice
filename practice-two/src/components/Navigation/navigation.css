.nav__menu {
  display: none;
}

.nav__list {
  list-style-type: none;
  display: flex;
  align-items: center;
  letter-spacing: 1px;

  font-family: "DM Sans";
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 27px;
}

.nav__item {
  margin: 0 32px;
}


.nav__item:last-child {
  margin-right: 0;
}

.active {
  text-decoration: underline;
}
  