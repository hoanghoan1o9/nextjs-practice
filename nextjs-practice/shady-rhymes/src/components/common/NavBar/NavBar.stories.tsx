// Libraries
import Storybook from '@storybook/react';

// Components
import NavBar from './index';

// Constants
import { NAV_ITEMS } from '@constants/navigation';

export default {
  title: 'Components/NavBar',
  component: NavBar,
} as Storybook.ComponentMeta<typeof NavBar>;

const Template: Storybook.ComponentStory<typeof NavBar> = (args) => (
  <NavBar {...args} />
);

export const DefaultNavBar = Template.bind({});
DefaultNavBar.args = {
  navItems: NAV_ITEMS,
};
