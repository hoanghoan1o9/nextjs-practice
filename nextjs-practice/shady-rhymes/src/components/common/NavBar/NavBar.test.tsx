// Libraries
import React from 'react';
import { cleanup, render } from '@testing-library/react';

// Components
import NavBar from './index';

import { NAV_ITEMS } from '@constants/navigation';

describe('NavBar render', () => {
  afterEach(cleanup);

  it('should render NavBar component', () => {
    const { getByTestId } = render(<NavBar navItems={NAV_ITEMS} />);

    const navBar = getByTestId('TestNavbar');
    expect(navBar).toBeTruthy();
    expect(navBar).toMatchSnapshot();
  });
});
