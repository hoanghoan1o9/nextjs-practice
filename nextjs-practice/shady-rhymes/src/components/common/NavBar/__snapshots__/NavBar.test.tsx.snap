// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render should render NavBar component 1`] = `
<nav
  class="css-1lekzkb"
  data-testid="TestNavbar"
>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 120px; height: 35px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
  >
    <img
      alt="main-logo"
      data-nimg="fixed"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
  <div
    class="chakra-stack css-4jkpek"
  >
    <div
      class="css-k008qs"
    >
      <a
        aria-controls="popover-content-:r1:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1y6ic72"
        href="/#"
        id="popover-trigger-:r1:"
      >
        Furniture  
      </a>
    </div>
    <div
      class="css-k008qs"
    >
      <a
        aria-controls="popover-content-:r3:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1y6ic72"
        href="/#"
        id="popover-trigger-:r3:"
      >
        Shop
      </a>
    </div>
    <div
      class="css-k008qs"
    >
      <a
        aria-controls="popover-content-:r5:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1y6ic72"
        href="/#"
        id="popover-trigger-:r5:"
      >
        About Us
      </a>
    </div>
    <div
      class="css-k008qs"
    >
      <a
        aria-controls="popover-content-:r7:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="chakra-link css-1y6ic72"
        href="/#"
        id="popover-trigger-:r7:"
      >
        Contact
      </a>
    </div>
  </div>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 24px; height: 24px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
  >
    <img
      alt="bag-icon"
      data-nimg="fixed"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
</nav>
`;
